core/templates/dev/head/domain/topic/TopicRightsObjectFactory.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * topic rights domain objects.
+ */
+
+oppia.factory('TopicRightsObjectFactory', [
+  function() {
+    var TopicRights = function(topicRightsObject) {
+      this._topicId = topicRightsObject.topic_id;
+      this._isPublished = topicRightsObject.is_published;
+      this._canEditTopic = topicRightsObject.can_edit_topic;
+    };
+
+    // Instance methods
+
+    TopicRights.prototype.getTopicId = function() {
+      return this._topicId;
+    };
+
+    TopicRights.prototype.getCanEditTopic = function() {
+      return this._canEditTopic;
+    };
+
+    TopicRights.prototype.getIsPublished = function() {
```

Comment:
topicRights.isPublished()

If you want to avoid collision with the backend dict / private names, consider using this._published and this._editableByCurrentUser instead in both of those.

Reply:
Done.